types/simulation/types.go

```diff
@@ -36,12 +41,14 @@ type Content interface {
 }
 
 type WeightedProposalMsg interface {
-	AppParamsKey() string           // key used to retrieve the value of the weight from the simulation application params
-	DefaultWeight() int             // default weight
-	MsgSimulatorFn() MsgSimulatorFn // msg simulator function
+	AppParamsKey() string            // key used to retrieve the value of the weight from the simulation application params
+	DefaultWeight() int              // default weight
+	MsgSimulatorFn() MsgSimulatorFnX // msg simulator function
 }
 
+// Deprecated: use MsgSimulatorFnX
 type MsgSimulatorFn func(r *rand.Rand, accs []Account, cdc address.Codec) (sdk.Msg, error)
```

Comment:
why not break this api? users will need to migrate anyways, we can break it safely

Reply:
The change is handled by an adapter. I had avoided breaking changes but there is no way around baseapp removal.